aram/src/main/java/fr/bmartel/aram/AccessRuleMaster.java

```diff
@@ -324,8 +324,10 @@ private void deleteArDo(byte[] buf) {
         short ofs = ISO7816.OFFSET_CDATA;
 
         checkTLV(buf, ofs, (byte) 0xF1, (short) (6 + RuleEntry.SIZE_AID + RuleEntry.SIZE_HASH + RuleEntry.SIZE_RULE));
-
-        if (buf[(short) (ofs + 2)] == (byte) 0x4F) {
+        if (buf[(short) (ofs + 1)] == 0) {
+            //delete all rules if length == 0
+            RuleEntry.deleteAll();
```

Comment:
Why not also ad an explicit return here, would make the else-if chain a bit easier to grasp?

Reply:
Allright with me, however RuleEntry.deleteAll(); hangs when more than 1 rule is added

Reply:
Is RuleEntry.deleteAll(); implemented correctly?